reader.go

```diff
@@ -30,33 +30,140 @@ func (o *ReaderOptions) norm(name string) {
 
 // Reader reads data from a remote feed.
 type Reader struct {
-	remote *bfs.Object
-	opt    ReaderOptions
-	ctx    context.Context
-	num    int
+	ctx context.Context
+	opt ReaderOptions
 
-	br io.ReadCloser // bfs reader
-	cr io.ReadCloser // compression reader
-	fd FormatDecoder
+	remotes  []*bfs.Object
+	cur      *streamReader
+	pos, num int
 }
 
 // NewReader inits a new reader.
 func NewReader(ctx context.Context, remote *bfs.Object, opt *ReaderOptions) (*Reader, error) {
+	return MultiReader(ctx, []*bfs.Object{remote}, opt), nil
+}
+
+// MultiReader inits a new reader for multiple remotes.  Remotes are read sequentially as if concatenated.
+// Once all remotes are fully read, Read will return EOF.
+func MultiReader(ctx context.Context, remotes []*bfs.Object, opt *ReaderOptions) *Reader {
 	var o ReaderOptions
 	if opt != nil {
 		o = *opt
 	}
-	o.norm(remote.Name())
+	if len(remotes) > 0 {
+		o.norm(remotes[0].Name())
+	}
 
 	return &Reader{
-		remote: remote,
-		opt:    o,
-		ctx:    ctx,
-	}, nil
+		remotes: remotes,
+		opt:     o,
+		ctx:     ctx,
+	}
 }
 
 // Read reads raw bytes from the feed.
 func (r *Reader) Read(p []byte) (int, error) {
+	if !r.ensureCurrent() {
+		return 0, io.EOF
+	}
+
+	n, err := r.cur.Read(p)
+	if err == io.EOF {
```

Comment:
Nit: I always suggest `errors.Is(err, io.EOF)` etc - looks to be the modern Go std for dealing with errors.